src/librustc/mir/interpret/value.rs

```diff
@@ -421,6 +421,12 @@ impl<'tcx, Tag> Scalar<Tag> {
         Ok(b as u8)
     }
 
+    pub fn to_u16(self) -> InterpResult<'static, u16> {
```

Comment:
While we're at it... could you sprinkle doc comments on the methods in this PR (or another...)?

Reply:
How about 
```    
// Consume the scalar to produce an `(u|i)XX`, it fails if the scalar is a pointer .
```

Reply:
New sentence at "It fails", otherwise sgtm.